### PR TITLE
GOLD-176: Multi-sig for network config change

### PR DESCRIPTION
**Pull request by kgmyatthu**
image
Complete with unit test. Backward compatible and configurable multi sig support for change config and change network params internal txs

### CODE REVIEW

File: src/setup/helpers.ts [hunk 1]
```diff
@@ -107,3 +108,38 @@ export function hashSignedObj(obj): string {
...
+
+    // The sig owner is listed on the server and
+    // The sig owner has enough security clearance
+    // The signature is valid
+    if(
+        allowedPubkeys[sigs[i].owner] && 
+        allowedPubkeys[sigs[i].owner] >= requiredSecurityLevel && 
+        crypto.verify(payload_hash, sigs[i].sig, sigs[i].owner)
+    ){
+      validSigs++
```
Comment (arhamj) [inline src/setup/helpers.ts:151]: If we reach minSigRequired we can exit this loop early.

Comment (kgmyatthu) [reply src/setup/helpers.ts:151]: This is done

File: src/setup/helpers.ts [hunk 2]
```diff
@@ -107,3 +108,41 @@ export function hashSignedObj(obj): string {
...
+/**
+@param rawPayload: any - The original payload stripped of the signatures
+@param sigs: Sign[] - The signatures to verify
+@param allowedPubkeys: {[pubkey: string]: DevSecurityLevel} - The public keys that are allowed to sign the payload
+@param minSigRequired: number - The minimum number of signatures required
+@param requiredSecurityLevel: DevSecurityLevel - The minimum security level required to sign the payload
+@returns boolean - True if the payload is signed by the required number of authorized public keys with the required security level
+**/
+export function verifyMultiSigs(
+  rawPayload: {},
```
Comment (S0naliThakur) [inline src/setup/helpers.ts:121]: In verifyMultiSigs, we do not check for duplicate signs in the payload, which could result in a logic bypass.

**Review by S0naliThakur — CHANGES_REQUESTED** [review]

File: src/setup/helpers.ts (hunk 2 above)
Comment (kgmyatthu) [reply src/setup/helpers.ts:121]: This is valid and changes are coming

**Pull request closed without merging**